skyfield/keplerlib.py

```diff
@@ -578,19 +609,15 @@ def kepler(x):
         lcount += 1
         not_done = (lower < x) * (x < upper) * (lcount < mostc)
 
-    c0, c1, c2, c3 = stumpff(f*x*x)
-    br = br0*c0 + x*(b2rv*c1 + x*(bq*c2))
+    c0, c1, c2, c3 = stumpff(x*x*f)
+    br = c0*br0 + x*(c1*b2rv + x*(c2*bq))
 
     pc = 1 - qovr0 * x**2 * c2
     vc = dt - bq * x**3 * c3
     pcdot = -qovr0 / br * x * c1
     vcdot = 1 - bq / br * x**2 * c2
 
-    if return_1d_array:
-        position_prop = pc*position + vc*velocity
-        velocity_prop = pcdot*position + vcdot*velocity
-    else:
-        position_prop = pc*position[newaxis].T + vc*velocity[newaxis].T
-        velocity_prop = pcdot*position[newaxis].T + vcdot*velocity[newaxis].T
+    position_prop = pc[newaxis, :, :]*position[:, :, newaxis] + vc[newaxis, :, :]*velocity[:, :, newaxis]
```

Comment:
I'm happy to see `if` statements disappearing!

Reply:
I could have removed that if statement a while ago thanks to `np.squeeze`!